File: java/visitor/DayShiftScheduleManagement.java
package visitor;

public class DayShiftScheduleManagement implements ScheduleManagement {
    @Override
    public void generateReport() {
        System.out.println("Generating report for day shift...");
    }

    @Override
    public void calculateOverTime() {
        System.out.println("Calculating over time for day shift..");
    }

    @Override
    public void accept(ScheduleManagementVisitor scheduleManagementVisitor) {
        scheduleManagementVisitor.visit(this);
    }

}

File: C#/Visitor/DayShiftScheduleManagement.cs
namespace Visitor;

public sealed class DayShiftScheduleManagement : IScheduleManagement
{
    public void GenerateReport()
        => Console.WriteLine("Generating report for day shift...");

    public void CalculateOverTime()
        => Console.WriteLine("Calculating over time for day shift..");

    public void Accept(IScheduleManagementVisitor scheduleManagementVisitor)
        => scheduleManagementVisitor.Visit(this);
}
